File: dcache/src/main/java/dora/rx/RxTransformer.java
package dora.rx;

import io.reactivex.Observable;
import io.reactivex.Observer;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.functions.Consumer;
import io.reactivex.schedulers.Schedulers;

public class RxTransformer {

    public static <T> void doApi(Observable<T> observable, Consumer<T> consumer) {
        observable.observeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .onErrorResumeNext(Observable.<T>empty())
                .subscribe(consumer);
    }

    public static <T> void doApi(Observable<T> observable, Observer<T> observer) {
        observable.observeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .onErrorResumeNext(Observable.<T>empty())
                .subscribe(observer);
    }
}

File: dcache/src/main/java/dora/http/rx/RxTransformer.kt
package dora.http.rx

import android.annotation.SuppressLint
import io.reactivex.Observable
import io.reactivex.Observer
import io.reactivex.schedulers.Schedulers
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.functions.Consumer

/**
 * RxJava transformer that provides utility methods for executing Rx functions.
 * 简体中文：RxJava变换器，提供执行Rx函数的工具方法。
 */
object RxTransformer {

    @SuppressLint("CheckResult")
    fun <M> doApiConsumer(observable: Observable<M>, consumer: Consumer<M>) {
        observable.subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .onErrorResumeNext(Observable.empty())
            .subscribe(consumer)
    }

    fun <M> doApiObserver(observable: Observable<M>, observer: Observer<M>) {
        observable.subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .onErrorResumeNext(Observable.empty())
            .subscribe(observer)
    }
}